src/routes/landing/components/FilterableCitySelector.js

```diff
@@ -17,34 +19,70 @@ type PropsType = {
 }
 
 type StateType = {
-  filterText: string
+  filterText: string,
+  currentLongitude: number | null,
+  currentLatitude: number | null,
+  renderLocationList: boolean
 }
 
 class FilterableCitySelector extends React.Component<PropsType, StateType> {
   constructor (props: PropsType) {
     super(props)
-    this.state = { filterText: '' }
+    this.state = {
+      filterText: '',
+      currentLongitude: null,
+      currentLatitude: null,
+      renderLocationList: false
+    }
   }
 
   onFilterTextChange = (filterText: string) => this.setState({ filterText })
 
+  _onPressLocationButton = () => {
+    if (Platform.OS === 'ios') {
+      Geolocation.requestAuthorization()
+    }
+    Geolocation.getCurrentPosition(
+      position => {
+        const currentLongitude = position.coords.longitude
+        const currentLatitude = position.coords.latitude
+        this.setState({ currentLongitude: currentLongitude })
+        this.setState({ currentLatitude: currentLatitude })
+      },
+      error => {
+        alert(error.message)
+      },
+      { enableHighAccuracy: false, timeout: 20000, maximumAge: 1000 }
+    )
+    this.setState({ renderLocationList: true })
+  }
+
   render () {
     const { cities, t, theme } = this.props
     const filterText = this.state.filterText
 
     return (
       <>
-        <SearchInput
+        <View style={{ flexDirection: 'row', justifyContent: 'space-between' }} >
+          <SearchInput style={{ flexGrow: 1 }}
           filterText={filterText}
           onFilterTextChange={this.onFilterTextChange}
           placeholderText={t('searchCity')}
           spaceSearch={false}
           theme={theme} />
+          <TouchableOpacity onPress={this._onPressLocationButton} style={{ flexGrow: 0 }}>
+            <Image source={
+              require('../../../../assets/app-locationButton.png')} />
```

Comment:
Just import this normally at the top of the file with an import.

Reply:
If I remember correctly we use (or used to use) `FastImage` instead of the default ReactNative `Image`.

Reply:
Correct, we have a custom Image component. See the Thumbnail component for the proper usage.